lib/active_merchant/billing/gateways/zuora.rb

```diff
@@ -0,0 +1,170 @@
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class ZuoraGateway < Gateway
+      self.test_url = 'https://rest.apisandbox.zuora.com/v1/'
+      self.live_url = 'https://rest.zuora.com/v1/'
+
+      self.supported_countries = ['AU']
+      self.default_currency = 'AUD'
+      self.supported_cardtypes = [:visa, :master, :american_express, :discover]
+
+      self.homepage_url = 'https://www.zuora.com/'
+      self.display_name = 'Zuora'
+
+      CARD_BRAND_MAP = {
+        'visa'              => 'Visa',
+        'mastercard'        => 'Mastercard',
+        'american_express'  => 'AmericanExpress',
+        'discover'          => 'Discover'
+      }.freeze
+
+      def initialize(options = {})
+        requires!(options, :username, :password)
+        super
+      end
+
+      def store(credit_card, options = {})
+        post = {}
+        add_address(post, options)
+        add_payment(post, credit_card, options)
+        add_store_options(post, options)
+        commit('accounts', post)
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(%r((apiAccessKeyId: )[^\s\\]+)i, '\1[FILTERED]').
+          gsub(%r((apiSecretAccessKey: )[^\s\\]+)i, '\1[FILTERED]').
+          gsub(%r((cardNumber\\\":\\\")\d+)i, '\1[FILTERED]').
+          gsub(%r((securityCode\\\":\\\")\d+)i, '\1[FILTERED]')
+      end
+
+      private
+
+      def add_address(post, options)
+        if (address = (options[:billing_address] || options[:address]))
+          post[:billToContact] = {
+            address1: address[:address1],
+            city: address[:city],
+            state: address[:state],
+            zipCode: address[:zip],
+            country: address[:country],
+            workEmail: options[:email],
+            firstName: options[:first_name],
+            lastName: options[:last_name]
+          }.reject { |_, v| v.blank? }
+        end
+      end
+
+      def add_payment(post, payment, options = {})
+        if payment.respond_to?(:number)
+          post[:currency] = options[:currency]
+          post[:creditCard] = {
+            cardNumber: payment.number,
+            cardType: CARD_BRAND_MAP.fetch(payment.brand),
+            expirationMonth: format(payment.month, :two_digits),
+            expirationYear: format(payment.year, :four_digits),
+            securityCode: payment.verification_value
+          }.reject { |_, v| v.blank? }
```

Comment:
Rails adds `#compact` to Hash if you wish to use it.

https://apidock.com/rails/Hash/compact

Reply:
Except it doesn’t remove blank items. Carry on.

Reply:
But ActiveMerchant doesn't necessarily has access to all the nice railties.